### PR TITLE
Updated deep learning framework integrations

### PR DESCRIPTION
**Pull request by jakubczakon**
* added "ANONYMOUS" token 
* updated with the additional functionality of `Pytorch-Lightning`
* updated with the additional checkpoint saving functionality of `NeptuneSaver` in `Ignite`
* updated with new `NeptuneLogger` for `Catalyst`

### CODE REVIEW

File: docs/integrations/catalyst.rst [hunk 1]
```diff
@@ -6,39 +6,46 @@ Log Catalyst metrics to neptune
 
 Prerequisites
 -------------
-Integration with |Catalyst| framework is introduced as a part of logging module so just need to have |neptune-client| installed.
+Integration with |Catalyst| framework is introduced as a part of as a part of logging module so just need to have |neptune-client| and |neptune-contrib| installed.
```
Comment (pitercl) [inline docs/integrations/catalyst.rst]: `as a part of as a part of`

File: docs/integrations/catalyst.rst [hunk 2]
```diff
@@ -159,8 +173,12 @@ Remember to change your credentials in the **runner.train()**:
 
 .. |example experiment| raw:: html
 
-    <a href="https://ui.neptune.ai/o/neptune-ai/org/catalyst-integration/e/CAT-6/charts" target="_blank">example experiment</a>
+    <a href="https://ui.neptune.ai/o/neptune-ai/org/catalyst-integration/e/CAT-13/charts" target="_blank">example experiment</a>
```
Comment (pitercl) [inline docs/integrations/catalyst.rst]: This exp doesn't exist.

Comment (jakubczakon) [reply docs/integrations/catalyst.rst]: done

File: docs/integrations/catalyst.rst (hunk 1 above)
Comment (jakubczakon) [reply docs/integrations/catalyst.rst]: fixed

**Pull request merged**